### Concurso predecir compra de casas.

En este modelo, eliminaremos algunas de las variables que no son significativas según la rueba de valor P hecha con anterioridad 

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, OneHotEncoder
import optuna 
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import models, layers 
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, Activation
from tensorflow.keras.optimizers import RMSprop, SGD, Adam
from tensorflow.keras import regularizers
import matplotlib.pyplot as plt
import statsmodels.api as sm

In [3]:
import mlflow
import mlflow.tensorflow
mlflow.tensorflow.autolog()
import plotly.io as pio

In [5]:
import os
from getpass import getpass

#vinculamos la cuenta de dagshub con mlflow
os.environ['MLFLOW_TRACKING_USERNAME'] = "mseguracarrillo7"
os.environ['MLFLOW_TRACKING_PASSWORD'] = getpass("Enter your DAGsHub access token or password: ")
os.environ["MLFLOW_TRACKING_URI"]="https://dagshub.com/mseguracarrillo7/Prediccion_CompraC"
mlflow.set_tracking_uri("https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow")

Enter your DAGsHub access token or password:  ········


In [72]:
data= pd.DataFrame(pd.read_csv("global_house_purchase_dataset.csv"))

In [74]:
paises_unicos = data['country'].unique() #países en la base de datos 

print("Los países en el dataset son:")
print(paises_unicos)

Los países en el dataset son:
['France' 'South Africa' 'Germany' 'Canada' 'Brazil' 'UAE' 'Australia'
 'UK' 'USA' 'China' 'Singapore' 'India' 'Japan']


In [76]:
# Tasas de conversión aproximadas a USD (Octubre 2025)
# 1 Moneda Local = X Dólares USD
rates_to_usd = {
    'France': 1.08,        # Euro (EUR)
    'South Africa': 0.053,   # Rand (ZAR)
    'Germany': 1.08,       # Euro (EUR)
    'Canada': 0.73,        # Dólar Canadiense (CAD)
    'Brazil': 0.18,        # Real Brasileño (BRL)
    'UAE': 0.27,           # Dírham de los EAU (AED)
    'Australia': 0.66,     # Dólar Australiano (AUD)
    'UK': 1.25,            # Libra Esterlina (GBP)
    'USA': 1.0,            # Dólar Estadounidense (USD)
    'China': 0.14,         # Yuan Chino (CNY)
    'Singapore': 0.74,     # Dólar de Singapur (SGD)
    'India': 0.012,        # Rupia India (INR)
    'Japan': 0.0064       # Yen Japonés (JPY)
}

In [78]:
data['price_USD'] = data.apply(
    lambda row: row['price'] * rates_to_usd.get(row['country'], 1.0), 
    axis=1
) #creamos una nueva columna de precio de la casa en la que aplicamos la conversión según el país 

#ahora haremos lo mismo para las columnas (customer_salary, loan_amount, monthly_expenses, down_payment )
data['customer_salary_USD'] = data.apply(
    lambda row: row['customer_salary'] * rates_to_usd.get(row['country'], 1.0), 
    axis=1
)

data['loan_amount_USD'] = data.apply(
    lambda row: row['loan_amount'] * rates_to_usd.get(row['country'], 1.0), 
    axis=1
)

data['monthly_expenses_USD'] = data.apply(
    lambda row: row['monthly_expenses'] * rates_to_usd.get(row['country'], 1.0), 
    axis=1
)

data['down_payment_USD'] = data.apply(
    lambda row: row['down_payment'] * rates_to_usd.get(row['country'], 1.0), 
    axis=1
)

#borramos las columnas iniciales 
data = data.drop(['price', 'customer_salary', 'loan_amount', 'monthly_expenses', 'down_payment'], axis=1)


Ahora borraremos algunas columnas que no tienen mucha influencia en el modelo, como la identificación que se le da a cada dato, la ciudad a la que pertenece la casa, los casos legales, baños y garage. 

In [45]:
x = data.drop(["decision", 'property_id', 'city', 'legal_cases_on_property', 'bathrooms', 'garage'],axis=1)

In [47]:
x = pd.get_dummies(x, drop_first=True) 

In [49]:
scaler = StandardScaler()
x_scaled = pd.DataFrame(scaler.fit_transform(x), columns=x.columns) #aplicamos la normalizaciómn a las variables en x

In [51]:
y = data['decision']

In [53]:
x_train, x_Test, y_train, y_Test = train_test_split(x_scaled,y, train_size=0.7, random_state=123)

In [55]:
x_test, x_val, y_test, y_val = train_test_split(x_Test,y_Test, train_size=2/3, random_state=123)

Buscaremos nuevos hiperparámetros en Optuna para este nuevo modelo.

In [58]:
def create_model(trial):
    with mlflow.start_run(nested=True): 
        params= {
            "learning_rate": trial.suggest_float("learning_rate",1e-4, 1e-2, log=True),
            "n_layers": trial.suggest_int("n_layers",1,3),
            "dropout_rate": trial.suggest_float("dropout_rate", 0.1, 0.5),
            "optimizer": trial.suggest_categorical("optimizer", ["adam", "rmsprop", "sgd"])
        }

        mlflow.log_params(params)
        model = models.Sequential()
        model.add(tf.keras.layers.Input(shape=(x_train.shape[1],))) 
    
        for i in range (params["n_layers"]):
            n_units = trial.suggest_int(f"n_units_l{i}", 32, 256)
            model.add(layers.Dense(n_units, activation = 'relu'))
            model.add(tf.keras.layers.Dropout(params["dropout_rate"])) #aplicamos el dropout sugerido
            
            mlflow.log_param(f"n_units_l{i}", n_units)
        model.add(layers.Dense(1, activation='sigmoid'))

        lr = params["learning_rate"]
        optimizer_selected = params["optimizer"]

        if optimizer_selected == "adam":
            optimizer = tf.keras.optimizers.Adam(learning_rate = lr)
        elif optimizer_selected == "sgd":
            optimizer = tf.keras.optimizers.SGD(learning_rate = lr)
        elif optimizer_selected == "rmsprop":
            optimizer = tf.keras.optimizers.RMSprop(learning_rate = lr)

    #usaremos binary cross entropy porque solo tenemos dos pposibles salidas (compro, no compro)
        model.compile(optimizer = optimizer,
                      loss = "binary_crossentropy",
                      metrics = ["accuracy"])
    return model 

mlflow.keras.autolog()

In [60]:
def objective(trial):
    modelo= create_model(trial)
    history= modelo.fit(x_train, y_train, epochs=5, batch_size=64, verbose=0)
    loss, accuracy = modelo.evaluate(x_test, y_test, verbose=0)
    mlflow.log_metric("accuracy", accuracy)
    return accuracy 

In [62]:
#prueba sin algunas variables.
study = optuna.create_study(direction = "maximize")
study.optimize(objective, n_trials=30)

[I 2025-10-10 20:19:01,891] A new study created in memory with name: no-name-60e0d80e-52e8-46a4-afd8-de3d7dc5616c


🏃 View run charming-fish-552 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/cc106d82551a48f686f2a291092c94d6
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:19:09 INFO mlflow.utils.autologging_utils: Created MLflow autologging run with ID '2cfafde4aab048e0af6a851ec2fc74da', which will track hyperparameters, performance metrics, model artifacts, and lineage information for the current keras workflow


🏃 View run gregarious-skink-240 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/2cfafde4aab048e0af6a851ec2fc74da
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:19:51 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:19:55,627] Trial 0 finished with value: 0.9225249886512756 and parameters: {'learning_rate': 0.00046825622495482296, 'n_layers': 2, 'dropout_rate': 0.19332813091419765, 'optimizer': 'rmsprop', 'n_units_l0': 111, 'n_units_l1': 98}. Best is trial 0 with value: 0.9225249886512756.


🏃 View run bemused-hen-628 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/344d8440e7e1402fa8edeb6dfa0c6c48
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:20:47 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:20:53,975] Trial 1 finished with value: 0.9211000204086304 and parameters: {'learning_rate': 0.0007909283596268475, 'n_layers': 3, 'dropout_rate': 0.2909808793517391, 'optimizer': 'rmsprop', 'n_units_l0': 193, 'n_units_l1': 56, 'n_units_l2': 200}. Best is trial 0 with value: 0.9225249886512756.


🏃 View run adorable-mole-513 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/04feee3c4c1f4dafb27183010d57d630
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:21:33 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:21:40,379] Trial 2 finished with value: 0.9233250021934509 and parameters: {'learning_rate': 0.004752470155527779, 'n_layers': 1, 'dropout_rate': 0.18917360449709486, 'optimizer': 'adam', 'n_units_l0': 214}. Best is trial 2 with value: 0.9233250021934509.


🏃 View run bittersweet-colt-207 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/386de7265cd04a3c87b282301c476504
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:22:27 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:22:33,836] Trial 3 finished with value: 0.8759250044822693 and parameters: {'learning_rate': 0.004368184530598986, 'n_layers': 2, 'dropout_rate': 0.37358366241815977, 'optimizer': 'sgd', 'n_units_l0': 250, 'n_units_l1': 49}. Best is trial 2 with value: 0.9233250021934509.


🏃 View run glamorous-finch-859 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/f849a51090564cef8d93d7e3d4e6c1b9
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:23:27 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:23:34,037] Trial 4 finished with value: 0.9239000082015991 and parameters: {'learning_rate': 0.0009144213144308646, 'n_layers': 3, 'dropout_rate': 0.21751914156023552, 'optimizer': 'adam', 'n_units_l0': 152, 'n_units_l1': 68, 'n_units_l2': 172}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run crawling-cow-808 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/e865ea0fa20443e09a51d2bfdc8f1dea
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:24:29 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:24:35,373] Trial 5 finished with value: 0.9225000143051147 and parameters: {'learning_rate': 0.002399311788807765, 'n_layers': 2, 'dropout_rate': 0.4512179662098389, 'optimizer': 'adam', 'n_units_l0': 43, 'n_units_l1': 93}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run dapper-gnu-519 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/0bc25c31a1a348ccb3b38b7b2bf01935
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:25:24 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:25:28,044] Trial 6 finished with value: 0.9223250150680542 and parameters: {'learning_rate': 0.003784065624632361, 'n_layers': 1, 'dropout_rate': 0.38438070280193937, 'optimizer': 'rmsprop', 'n_units_l0': 221}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run rumbling-quail-508 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/e5d062905d3f4a7aac16d3e22a38ab69
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:26:16 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:26:20,622] Trial 7 finished with value: 0.8668000102043152 and parameters: {'learning_rate': 0.002461930341765788, 'n_layers': 2, 'dropout_rate': 0.22961919515794468, 'optimizer': 'sgd', 'n_units_l0': 118, 'n_units_l1': 172}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run upbeat-frog-388 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/9c5d171bd01b4703a210f59d66d61aed
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:28:28 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:28:36,429] Trial 8 finished with value: 0.770550012588501 and parameters: {'learning_rate': 0.000743224925887533, 'n_layers': 3, 'dropout_rate': 0.3366795557982888, 'optimizer': 'sgd', 'n_units_l0': 78, 'n_units_l1': 36, 'n_units_l2': 49}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run bouncy-duck-214 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/8eaf8b18cb4f4b098f85c3a2f9d1ca57
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:29:31 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:29:38,177] Trial 9 finished with value: 0.8756250143051147 and parameters: {'learning_rate': 0.0037104603019055406, 'n_layers': 3, 'dropout_rate': 0.24810358038599978, 'optimizer': 'sgd', 'n_units_l0': 119, 'n_units_l1': 129, 'n_units_l2': 78}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run bold-mink-37 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/0e0c4547a5624bab862ebc9ad0e1bbc0
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:31:25 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:31:32,070] Trial 10 finished with value: 0.9236000180244446 and parameters: {'learning_rate': 0.00016076179555551972, 'n_layers': 3, 'dropout_rate': 0.10459236970015894, 'optimizer': 'adam', 'n_units_l0': 169, 'n_units_l1': 242, 'n_units_l2': 256}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run rumbling-shrike-547 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/1d142665569c4e12a9e407f7d42405fa
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:35:48 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:35:53,886] Trial 11 finished with value: 0.9232000112533569 and parameters: {'learning_rate': 0.00013414573896121127, 'n_layers': 3, 'dropout_rate': 0.10545270034153953, 'optimizer': 'adam', 'n_units_l0': 166, 'n_units_l1': 254, 'n_units_l2': 255}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run lyrical-snipe-72 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/21ac85ffe13e4a4f81edf11cc37b9359
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:37:20 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:37:27,191] Trial 12 finished with value: 0.9226750135421753 and parameters: {'learning_rate': 0.00011802847959311247, 'n_layers': 3, 'dropout_rate': 0.10832561983112868, 'optimizer': 'adam', 'n_units_l0': 163, 'n_units_l1': 216, 'n_units_l2': 157}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run fortunate-bear-94 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/79dd32f37dc74f2384ac76dfcf2f2cc2
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:39:05 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:39:12,023] Trial 13 finished with value: 0.9236249923706055 and parameters: {'learning_rate': 0.00037677275686832197, 'n_layers': 3, 'dropout_rate': 0.1523290399569831, 'optimizer': 'adam', 'n_units_l0': 150, 'n_units_l1': 177, 'n_units_l2': 255}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run intelligent-foal-226 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/0d4c08c0276740e794547e0133d9c337
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:40:18 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:40:24,434] Trial 14 finished with value: 0.9237250089645386 and parameters: {'learning_rate': 0.0003392928447314029, 'n_layers': 3, 'dropout_rate': 0.17036938145094524, 'optimizer': 'adam', 'n_units_l0': 138, 'n_units_l1': 181, 'n_units_l2': 188}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run sneaky-donkey-472 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/141ca0852dea44408e294984aeb23541
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:41:18 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:41:24,949] Trial 15 finished with value: 0.9219750165939331 and parameters: {'learning_rate': 0.00028215819583830094, 'n_layers': 2, 'dropout_rate': 0.27157941699337096, 'optimizer': 'adam', 'n_units_l0': 86, 'n_units_l1': 158}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run fearless-wolf-752 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/4f1495f69fd6496c95c898d91deb2ca3
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:42:44 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:42:50,455] Trial 16 finished with value: 0.9230250120162964 and parameters: {'learning_rate': 0.0015400316617153976, 'n_layers': 3, 'dropout_rate': 0.18044143053564898, 'optimizer': 'adam', 'n_units_l0': 138, 'n_units_l1': 207, 'n_units_l2': 163}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run skillful-grub-826 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/75590433d537446fa30e928706dde06b
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:43:45 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:43:51,294] Trial 17 finished with value: 0.9237250089645386 and parameters: {'learning_rate': 0.0012437835992663675, 'n_layers': 1, 'dropout_rate': 0.22886654411989715, 'optimizer': 'adam', 'n_units_l0': 189}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run righteous-cow-799 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/74a01e29651f461e8e1bbfc315241fe9
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:44:47 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:44:53,232] Trial 18 finished with value: 0.9220250248908997 and parameters: {'learning_rate': 0.009253327112371146, 'n_layers': 3, 'dropout_rate': 0.15045025587045843, 'optimizer': 'adam', 'n_units_l0': 83, 'n_units_l1': 123, 'n_units_l2': 195}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run sneaky-gnu-63 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/855477ab238b4c788c57cc3ac580d9f8
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:45:48 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:45:54,265] Trial 19 finished with value: 0.9212250113487244 and parameters: {'learning_rate': 0.0002644814652294988, 'n_layers': 2, 'dropout_rate': 0.3446551390731705, 'optimizer': 'rmsprop', 'n_units_l0': 135, 'n_units_l1': 83}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run loud-bug-821 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/720ab4de5c1647e3b865e847a89b1045
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:46:46 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:46:52,129] Trial 20 finished with value: 0.921625018119812 and parameters: {'learning_rate': 0.0005386762481134216, 'n_layers': 2, 'dropout_rate': 0.49939149411144146, 'optimizer': 'adam', 'n_units_l0': 33, 'n_units_l1': 193}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run beautiful-bat-180 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/bc22ffe6f3c54e9bb1607d1f15523f21
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:47:42 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:47:48,584] Trial 21 finished with value: 0.9232749938964844 and parameters: {'learning_rate': 0.0013404571023816867, 'n_layers': 1, 'dropout_rate': 0.22007556556983016, 'optimizer': 'adam', 'n_units_l0': 187}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run worried-sheep-680 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/4d7c10623709491d94334493b061439e
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:48:39 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:48:43,132] Trial 22 finished with value: 0.9238749742507935 and parameters: {'learning_rate': 0.0010526669450973227, 'n_layers': 1, 'dropout_rate': 0.25930974017115105, 'optimizer': 'adam', 'n_units_l0': 193}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run intrigued-slug-675 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/6d0fb2508b2d40c784ad05ec976142d1
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:49:30 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:49:36,985] Trial 23 finished with value: 0.9232500195503235 and parameters: {'learning_rate': 0.0007966519145275216, 'n_layers': 1, 'dropout_rate': 0.28022633957337834, 'optimizer': 'adam', 'n_units_l0': 213}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run gaudy-conch-1000 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/ed5fb4840d8c4a20892484dc7a6cdb30
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:50:26 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:50:31,944] Trial 24 finished with value: 0.9232249855995178 and parameters: {'learning_rate': 0.00024124602049787705, 'n_layers': 2, 'dropout_rate': 0.15431797067112074, 'optimizer': 'adam', 'n_units_l0': 249, 'n_units_l1': 66}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run resilient-rat-797 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/517f628dbb2b401c88058f554ee56e78
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:51:36 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:51:40,995] Trial 25 finished with value: 0.9237750172615051 and parameters: {'learning_rate': 0.0005509329961665922, 'n_layers': 3, 'dropout_rate': 0.31221875279062045, 'optimizer': 'adam', 'n_units_l0': 150, 'n_units_l1': 141, 'n_units_l2': 113}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run skillful-wasp-376 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/dd319ad971834a4b82c7dcd5c966512f
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:52:28 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:52:32,153] Trial 26 finished with value: 0.9231500029563904 and parameters: {'learning_rate': 0.0020832118135964686, 'n_layers': 1, 'dropout_rate': 0.30948862413436107, 'optimizer': 'adam', 'n_units_l0': 172}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run enchanting-lynx-336 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/53a814fe509d4c9189c25b600d230119
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:53:27 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:53:33,630] Trial 27 finished with value: 0.9237750172615051 and parameters: {'learning_rate': 0.0005595973706178091, 'n_layers': 2, 'dropout_rate': 0.33762868934472257, 'optimizer': 'adam', 'n_units_l0': 229, 'n_units_l1': 117}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run colorful-rook-809 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/082ba06ad0144b33962c06c24a87ecc0
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:54:25 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:54:32,221] Trial 28 finished with value: 0.9207000136375427 and parameters: {'learning_rate': 0.001040193555171342, 'n_layers': 3, 'dropout_rate': 0.25738648172087436, 'optimizer': 'rmsprop', 'n_units_l0': 199, 'n_units_l1': 145, 'n_units_l2': 114}. Best is trial 4 with value: 0.9239000082015991.


🏃 View run melodic-dove-989 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/87ebc86cfa1948e9b99cd2923aaf6b31
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 20:55:26 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 20:55:33,843] Trial 29 finished with value: 0.7955999970436096 and parameters: {'learning_rate': 0.0004835355658049402, 'n_layers': 1, 'dropout_rate': 0.31218784652007325, 'optimizer': 'sgd', 'n_units_l0': 101}. Best is trial 4 with value: 0.9239000082015991.


No podemos esperar un muy buen resultado como el de la primera prueba, entonces no ha favorecido al modelo quitar las variables que hemos quitado, podemos probar con otras, aunque primero veamos el R-squared.

In [66]:
#creación del modelo
x_train = sm.add_constant(x_train, prepend=True)
modelo = sm.Logit(endog = y_train, exog = x_train)
modelo_fit = modelo.fit()
print(modelo_fit.summary())

Optimization terminated successfully.
         Current function value: 0.280878
         Iterations 8
                           Logit Regression Results                           
Dep. Variable:               decision   No. Observations:               140000
Model:                          Logit   Df Residuals:                   139965
Method:                           MLE   Df Model:                           34
Date:                Fri, 10 Oct 2025   Pseudo R-squ.:                  0.4803
Time:                        20:56:24   Log-Likelihood:                -39323.
converged:                       True   LL-Null:                       -75658.
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                       coef    std err          z      P>|z|      [0.025      0.975]
----------------------------------------------------------------------------------------------------
const                               -2.7946      0.017   -166.68

Tenemos un R-squared de 0.48 lo cual no es muy bueno, probemos con otras variables

In [107]:
x = data.drop(["decision", 'property_id', 'city'],axis=1)

In [109]:
x = pd.get_dummies(x, drop_first=True) 

In [111]:
scaler = StandardScaler()
x_scaled = pd.DataFrame(scaler.fit_transform(x), columns=x.columns) #aplicamos la normalizaciómn a las variables en x

In [113]:
y= data['decision']

In [115]:
x_train, x_Test, y_train, y_Test = train_test_split(x_scaled,y, train_size=0.7, random_state=123)
x_test, x_val, y_test, y_val = train_test_split(x_Test,y_Test, train_size=2/3, random_state=123)

In [102]:
#creación del modelo
x_train = sm.add_constant(x_train, prepend=True)
modelo = sm.Logit(endog = y_train, exog = x_train)
modelo_fit = modelo.fit()
print(modelo_fit.summary())

         Current function value: 0.128551
         Iterations: 35


C:\Users\Moises\anaconda3\envs\tf_env\lib\site-packages\statsmodels\base\model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


                           Logit Regression Results                           
Dep. Variable:               decision   No. Observations:               140000
Model:                          Logit   Df Residuals:                   139962
Method:                           MLE   Df Model:                           37
Date:                Fri, 10 Oct 2025   Pseudo R-squ.:                  0.7621
Time:                        21:00:39   Log-Likelihood:                -17997.
converged:                      False   LL-Null:                       -75658.
Covariance Type:            nonrobust   LLR p-value:                     0.000
                                       coef    std err          z      P>|z|      [0.025      0.975]
----------------------------------------------------------------------------------------------------
const                              -13.9002   1.76e+05  -7.89e-05      1.000   -3.45e+05    3.45e+05
property_size_sqft                   0.6130      0.024     25.615

Mejoró bastante el R-squared (0.7621), es un buen modelo, tiene el mismo valor que considerando todas las variables, entonces probemos entrenar la red nneuronal con estas.

In [117]:
#prueba sin algunas variables.
study = optuna.create_study(direction = "maximize")
study.optimize(objective, n_trials=30)

[I 2025-10-10 21:04:32,569] A new study created in memory with name: no-name-0de4651f-fd18-4e69-9dc3-d291b9461bfa


🏃 View run welcoming-turtle-148 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/3ed36442b47943fea16723ada1bb645d
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:05:51 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:05:56,453] Trial 0 finished with value: 0.9503250122070312 and parameters: {'learning_rate': 0.0045805424347809915, 'n_layers': 2, 'dropout_rate': 0.3157175385945594, 'optimizer': 'sgd', 'n_units_l0': 145, 'n_units_l1': 243}. Best is trial 0 with value: 0.9503250122070312.


🏃 View run amazing-gnat-169 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/f8cf65d242a94f8a98c9b9407bb9c519
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:06:52 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:06:57,291] Trial 1 finished with value: 0.9929749965667725 and parameters: {'learning_rate': 0.0001586577057763614, 'n_layers': 2, 'dropout_rate': 0.24811371324293852, 'optimizer': 'rmsprop', 'n_units_l0': 121, 'n_units_l1': 94}. Best is trial 1 with value: 0.9929749965667725.


🏃 View run agreeable-cat-862 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/b0e4a16690964ec09004d1ffe3a4ccb9
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:07:50 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:07:56,796] Trial 2 finished with value: 0.9983000159263611 and parameters: {'learning_rate': 0.0009600485672929747, 'n_layers': 1, 'dropout_rate': 0.1732749616023112, 'optimizer': 'rmsprop', 'n_units_l0': 109}. Best is trial 2 with value: 0.9983000159263611.


🏃 View run learned-shark-348 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/5fc7f3c3a1a04e49b21dcc2fe185e926
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:09:15 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:09:20,442] Trial 3 finished with value: 0.9974750280380249 and parameters: {'learning_rate': 0.002739064362112621, 'n_layers': 2, 'dropout_rate': 0.4336538054488779, 'optimizer': 'rmsprop', 'n_units_l0': 84, 'n_units_l1': 191}. Best is trial 2 with value: 0.9983000159263611.


🏃 View run painted-snipe-180 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/0b9d5674c9824c7b9a4d371d10fcf530
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:10:06 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:10:13,073] Trial 4 finished with value: 0.9177500009536743 and parameters: {'learning_rate': 0.001236275048041862, 'n_layers': 1, 'dropout_rate': 0.37919987058126237, 'optimizer': 'sgd', 'n_units_l0': 90}. Best is trial 2 with value: 0.9983000159263611.


🏃 View run marvelous-robin-994 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/cb96281d64734b1f90e88b4299f593ee
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:11:16 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:11:23,014] Trial 5 finished with value: 0.775950014591217 and parameters: {'learning_rate': 0.0008286792769866569, 'n_layers': 2, 'dropout_rate': 0.44818194468362427, 'optimizer': 'sgd', 'n_units_l0': 87, 'n_units_l1': 38}. Best is trial 2 with value: 0.9983000159263611.


🏃 View run hilarious-bug-59 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/382024045cd34d3286046d9d2c3d98b1
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:13:05 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:13:12,100] Trial 6 finished with value: 0.9983500242233276 and parameters: {'learning_rate': 0.00029718153096312366, 'n_layers': 3, 'dropout_rate': 0.27704846596796606, 'optimizer': 'adam', 'n_units_l0': 104, 'n_units_l1': 141, 'n_units_l2': 250}. Best is trial 6 with value: 0.9983500242233276.


🏃 View run delightful-frog-371 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/f135d2c86d034ab0b6c61361875312b8
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:14:40 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:14:46,491] Trial 7 finished with value: 0.9988499879837036 and parameters: {'learning_rate': 0.005130577652303885, 'n_layers': 2, 'dropout_rate': 0.49334236200747184, 'optimizer': 'rmsprop', 'n_units_l0': 223, 'n_units_l1': 88}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run skillful-roo-662 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/d460ee555b8f424bad64950744c79e96
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:15:39 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:15:46,591] Trial 8 finished with value: 0.9981750249862671 and parameters: {'learning_rate': 0.0029695442221887233, 'n_layers': 1, 'dropout_rate': 0.10773080442230892, 'optimizer': 'rmsprop', 'n_units_l0': 202}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run colorful-shoat-696 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/8a96541e514b4632a1c35634eaf43fc2
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:16:37 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:16:42,409] Trial 9 finished with value: 0.9986249804496765 and parameters: {'learning_rate': 0.002302287154731579, 'n_layers': 1, 'dropout_rate': 0.30656435095865364, 'optimizer': 'rmsprop', 'n_units_l0': 236}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run legendary-wasp-856 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/09066c9b3036418fbff907eaad8a5f7e
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:17:50 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:17:56,991] Trial 10 finished with value: 0.9931750297546387 and parameters: {'learning_rate': 0.007214948863310746, 'n_layers': 3, 'dropout_rate': 0.4935817754894527, 'optimizer': 'adam', 'n_units_l0': 34, 'n_units_l1': 67, 'n_units_l2': 42}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run honorable-wren-913 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/cff38690eea8444cb202a71ce03fc50a
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:18:52 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:18:57,973] Trial 11 finished with value: 0.9984250068664551 and parameters: {'learning_rate': 0.0019239997358966785, 'n_layers': 1, 'dropout_rate': 0.3557709080985675, 'optimizer': 'rmsprop', 'n_units_l0': 254}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run bald-snake-108 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/8951faa505184628bd0fc283a1ae91ba
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:20:23 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:20:29,740] Trial 12 finished with value: 0.9973750114440918 and parameters: {'learning_rate': 0.009049416435644433, 'n_layers': 3, 'dropout_rate': 0.21462881827758473, 'optimizer': 'rmsprop', 'n_units_l0': 249, 'n_units_l1': 126, 'n_units_l2': 152}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run kindly-perch-328 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/783daa286b314f1aa3b7f6d3a8c63931
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:21:25 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:21:31,652] Trial 13 finished with value: 0.9976249933242798 and parameters: {'learning_rate': 0.0047052677497605045, 'n_layers': 1, 'dropout_rate': 0.37270966774075986, 'optimizer': 'rmsprop', 'n_units_l0': 199}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run unique-hawk-304 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/77e561eb10b247da98cbc3473fc905c1
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:23:06 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:23:12,446] Trial 14 finished with value: 0.9980999827384949 and parameters: {'learning_rate': 0.0004139488664794371, 'n_layers': 2, 'dropout_rate': 0.49225238362827656, 'optimizer': 'rmsprop', 'n_units_l0': 201, 'n_units_l1': 195}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run serious-goat-330 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/c4235d6689e24a7cbc69c8e74604a688
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:24:39 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:24:45,442] Trial 15 finished with value: 0.9985250234603882 and parameters: {'learning_rate': 0.0017722366152670144, 'n_layers': 2, 'dropout_rate': 0.16185130387962404, 'optimizer': 'adam', 'n_units_l0': 169, 'n_units_l1': 94}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run grandiose-skink-760 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/9664bb4ad39341bd83802ea013a6b18c
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:25:48 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:25:54,245] Trial 16 finished with value: 0.9982249736785889 and parameters: {'learning_rate': 0.00451537909931885, 'n_layers': 1, 'dropout_rate': 0.3258124544184451, 'optimizer': 'rmsprop', 'n_units_l0': 223}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run delicate-gnat-530 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/8282e56c33ca4e9890559c23a7db3d60
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:27:21 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:27:26,668] Trial 17 finished with value: 0.9978749752044678 and parameters: {'learning_rate': 0.0005328882354699232, 'n_layers': 3, 'dropout_rate': 0.42326483838184176, 'optimizer': 'rmsprop', 'n_units_l0': 225, 'n_units_l1': 37, 'n_units_l2': 38}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run crawling-hound-843 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/a8ef3a902ef749078619fbc8e43020da
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:28:21 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:28:28,163] Trial 18 finished with value: 0.9986249804496765 and parameters: {'learning_rate': 0.002493005178727679, 'n_layers': 1, 'dropout_rate': 0.2633589927448025, 'optimizer': 'adam', 'n_units_l0': 165}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run illustrious-carp-106 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/0e6ede8c6d3a4e86bc145ca7556e0a19
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:30:15 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:30:22,842] Trial 19 finished with value: 0.9554499983787537 and parameters: {'learning_rate': 0.005809941863304696, 'n_layers': 2, 'dropout_rate': 0.3994190284646287, 'optimizer': 'sgd', 'n_units_l0': 231, 'n_units_l1': 184}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run nervous-eel-306 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/731d856249714b368dd34e0ee08f8d88
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:31:59 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:32:16,419] Trial 20 finished with value: 0.9959250092506409 and parameters: {'learning_rate': 0.003419024768871083, 'n_layers': 2, 'dropout_rate': 0.21434930480490416, 'optimizer': 'rmsprop', 'n_units_l0': 181, 'n_units_l1': 108}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run resilient-koi-976 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/2d264e855a3e48a19071b145f9fcecc0
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:33:08 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:33:14,474] Trial 21 finished with value: 0.99795001745224 and parameters: {'learning_rate': 0.0018305814232540366, 'n_layers': 1, 'dropout_rate': 0.27137625236341845, 'optimizer': 'adam', 'n_units_l0': 170}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run resilient-seal-617 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/8ffb576928b246f8ac6503f8f8a27eaf
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:34:07 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:34:13,858] Trial 22 finished with value: 0.9977999925613403 and parameters: {'learning_rate': 0.009880516763828034, 'n_layers': 1, 'dropout_rate': 0.33167863869364334, 'optimizer': 'adam', 'n_units_l0': 240}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run nervous-shad-318 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/39503462bceb4c3c8c8f4d03bebffbac
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:35:13 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:35:20,327] Trial 23 finished with value: 0.998199999332428 and parameters: {'learning_rate': 0.002340354202120363, 'n_layers': 1, 'dropout_rate': 0.2378134144495807, 'optimizer': 'adam', 'n_units_l0': 145}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run able-bug-208 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/871553a84cbc46019e2ab438367b0e4b
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:36:15 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:36:21,557] Trial 24 finished with value: 0.9986000061035156 and parameters: {'learning_rate': 0.001321428303714743, 'n_layers': 1, 'dropout_rate': 0.28216790023072863, 'optimizer': 'adam', 'n_units_l0': 213}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run silent-roo-574 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/1f547fdab6ff4c6a8883f2e078b300a8
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:37:16 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:37:21,159] Trial 25 finished with value: 0.9983500242233276 and parameters: {'learning_rate': 0.0007010081077187591, 'n_layers': 1, 'dropout_rate': 0.15364290198341685, 'optimizer': 'adam', 'n_units_l0': 181}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run rumbling-kite-431 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/b27b2e783cdc45199ecf5cb47ba56565
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:38:14 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:38:19,218] Trial 26 finished with value: 0.9983999729156494 and parameters: {'learning_rate': 0.0033068301035934513, 'n_layers': 1, 'dropout_rate': 0.45783188460178303, 'optimizer': 'rmsprop', 'n_units_l0': 59}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run grandiose-shrimp-535 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/7e4a63337d7944f8a9ca8ba043699b1c
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:39:47 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:39:52,710] Trial 27 finished with value: 0.9974750280380249 and parameters: {'learning_rate': 0.006426233904010512, 'n_layers': 2, 'dropout_rate': 0.20236444550947275, 'optimizer': 'rmsprop', 'n_units_l0': 158, 'n_units_l1': 166}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run secretive-perch-304 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/57b5c85886844e1dac502eeaa78c0925
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:40:48 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:40:57,638] Trial 28 finished with value: 0.9396250247955322 and parameters: {'learning_rate': 0.0014924038854799858, 'n_layers': 2, 'dropout_rate': 0.29560112777708364, 'optimizer': 'sgd', 'n_units_l0': 192, 'n_units_l1': 66}. Best is trial 7 with value: 0.9988499879837036.


🏃 View run valuable-sheep-497 at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0/runs/8549752f58df4397aad09bb2f3e3970a
🧪 View experiment at: https://dagshub.com/mseguracarrillo7/Prediccion_CompraC.mlflow/#/experiments/0


2025/10/10 21:42:33 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INTERNAL_ERROR: Response: {'error': 'unsupported endpoint, please contact support@dagshub.com'}
[I 2025-10-10 21:42:40,590] Trial 29 finished with value: 0.9976249933242798 and parameters: {'learning_rate': 0.004241528493577921, 'n_layers': 3, 'dropout_rate': 0.33816247993035625, 'optimizer': 'adam', 'n_units_l0': 131, 'n_units_l1': 145, 'n_units_l2': 249}. Best is trial 7 with value: 0.9988499879837036.


Ahora, probaremos con los mejores hiperparámetros obtenidos.

- learning rate = 0.0051305776
- n_layers= 2
- dropout rate = 0.49334236200747184
- optimizer = RMSprop
- n_unitsl0 = 223
- n_unitsl1 = 88

In [123]:
modelf = tf.keras.models.Sequential()
modelf.add(tf.keras.layers.Input(shape=(x_train.shape[1],)))
modelf.add(tf.keras.layers.Dense(223, activation='relu'))
modelf.add(tf.keras.layers.Dropout(0.49334236200747184))
modelf.add(tf.keras.layers.Dense(88, activation='relu'))
modelf.add(tf.keras.layers.Dropout(0.49334236200747184))
modelf.add(tf.keras.layers.Dense(1, activation='sigmoid'))

optimizer = tf.keras.optimizers.RMSprop(learning_rate= 0.0051305776)

modelf.compile(optimizer=optimizer, loss='binary_crossentropy', metrics=['accuracy'])
modelf.summary()

#entrenamos el modelo con el conjunto de entrenamiento
history= modelf.fit(x_train,y_train, epochs=40, batch_size=64, verbose=1, validation_data=(x_test, y_test))

Model: "sequential_61"

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┓
┃ Layer (type)                    ┃ Output Shape           ┃       Param # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━┩
│ dense_178 (Dense)               │ (None, 223)            │         8,697 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dropout_117 (Dropout)           │ (None, 223)            │             0 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dense_179 (Dense)               │ (None, 88)             │        19,712 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dropout_118 (Dropout)           │ (None, 88)             │             0 │
├─────────────────────────────────┼────────────────────────┼───────────────┤
│ dense_180 (Dense)               │ (None, 1)              │            89 │
└─────────────────────────────────┴────────────────────────┴───────────────┘

 Total params: 28,498 (111.32 KB)

 Trainable params: 28,498 (111.32 KB)

 Non-trainable params: 0 (0.00 B)

Epoch 1/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 24s 8ms/step - accuracy: 0.9742 - loss: 0.0639 - val_accuracy: 0.9972 - val_loss: 0.0075
Epoch 2/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 17s 8ms/step - accuracy: 0.9949 - loss: 0.0163 - val_accuracy: 0.9973 - val_loss: 0.0068
Epoch 3/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 16s 7ms/step - accuracy: 0.9961 - loss: 0.0128 - val_accuracy: 0.9980 - val_loss: 0.0057
Epoch 4/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 16s 7ms/step - accuracy: 0.9965 - loss: 0.0120 - val_accuracy: 0.9978 - val_loss: 0.0051
Epoch 5/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 21s 7ms/step - accuracy: 0.9966 - loss: 0.0121 - val_accuracy: 0.9983 - val_loss: 0.0043
Epoch 6/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 15s 7ms/step - accuracy: 0.9970 - loss: 0.0110 - val_accuracy: 0.9974 - val_loss: 0.0067
Epoch 7/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 15s 7ms/step - accuracy: 0.9968 - loss: 0.0115 - val_accuracy: 0.9981 - val_loss: 0.0049
Epoch 8/40
2188/2188 ━━━━━━━━━━━━━━━━━━━━ 16s 7ms/step - accuracy: 0.9971 - loss: 0

2025/10/11 19:46:06 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during keras autologging: INVALID_PARAMETER_VALUE: Response: {'error_code': 'INVALID_PARAMETER_VALUE'}


In [125]:
val_loss, val_accuracy = modelf.evaluate(x_val, y_val)
print(f"Precisión final: {val_accuracy*100:.2f}%")

625/625 ━━━━━━━━━━━━━━━━━━━━ 3s 4ms/step - accuracy: 0.9981 - loss: 0.0056
Precisión final: 99.81%


Tenemos una precisión un poco menor que en nuestra primera red, por lo tanto mi resultado principal será el de la primera con un 99.91% de precisión.